File: examples/TranslateMe.java
import subExamples.*;

public class TranslateMe {
	public static void main(String[] args){
		HelloWorld hw = new HelloWorld();
		System.out.println(hw.toString());


		HelloUniverse hu = new HelloUniverse();
		System.out.println(hu.toString());
		
			Object o = (Object)hu;
			System.out.println(o.toString());
		
		System.out.println(hu.createWorld());

		

	}
}

File: cppExamples/TranslateMe.cc
#include <iostream>
#include "java_lang.h"
#include "output.h"

using namespace java::lang;
using namespace subExamples;
using namespace std;

int main(void){

	HelloWorld hw = new __HelloWorld();
	cout << hw->__vptr->toString(hw)->data <<endl;

	HelloUniverse hu = new __HelloUniverse();
	cout << hu->__vptr->toString(hu)->data <<endl;

	Object o = (Object)hu;
	cout << o->__vptr->toString((Object)hu)->data <<endl;
	cout << hu->__vptr->createWorld(hu) <<endl;

	return 0;
}
